### PR TITLE
use versioneer to manage version numbers

### PR DESCRIPTION
**Pull request by jenshnielsen**
This means that if you install from a non release version you will get version numbers such as 0.4.0+30.g160bfbc45 where 0.4.0 is the last release before your current version. 30 are the number of commits since the release. g indicates that we use git and 160bfbc45 is the short commit hash.

This also means that there is no longer any need to manually update the version number in version.py 

versioneer.py and _version.py are auto generated and it probably does not make sense to review these files

Replaces #1631

### CODE REVIEW

**Comment by jenshnielsen** [comment]
Should probably exclude this code from codecov

File: qcodes/__init__.py [hunk 1]
```diff
@@ -115,4 +115,8 @@ def test(**kwargs):
     return retcode
 
 
-test.__test__ = False # type: ignore # Don't try to run this method as a test
+test.__test__ = False  # type: ignore # Don't try to run this method as a test
+
+from ._version import get_versions
```
Comment (astafan8) [inline qcodes/__init__.py]: why was it needed for duplicate code? isn't `from . import version \n __version__ = version.__version__` enough?

File: versioneer.py [hunk 2]
```diff
@@ -0,0 +1,1822 @@
+
+# Version: 0.18
```
Comment (astafan8) [inline versioneer.py]: will we ever need to upgrade this?))

**Comment by astafan8** [comment]
also, could you update the "how to release wiki" that "manually bumping the version is not needed" after this is merged?)

File: versioneer.py (hunk 2 above)
Comment (jenshnielsen) [reply versioneer.py]: Yes its documented how to upgrade here https://github.com/warner/python-versioneer#updating-versioneer 

I will add a link

**Comment by jenshnielsen** [comment]
@astafan8 yes the wiki is on my todo list after this is merged.

File: qcodes/__init__.py (hunk 1 above)
Comment (jenshnielsen) [reply qcodes/__init__.py]: Sure

File: versioneer.py (hunk 2 above)
Comment (jenshnielsen) [reply versioneer.py]: It's already documented in the doc string below. I think this is ok

Comment (astafan8) [reply versioneer.py]: yes, it is perfect! thanks for pointing me there

**Pull request merged**